.github/workflows/validate-pipelines.yml

```diff
@@ -0,0 +1,60 @@
+name: Validate Deployment Pipelines
+
+on:
+  pull_request:
+  push:
+      branches: [main, test-me-*]
+
+concurrency:
+    group: ${{ github.workflow }}-${{ github.head_ref || github.run_id }}
+    cancel-in-progress: true
+
+jobs:
+    files-changed:
+        name: files-changed
+        runs-on: ubuntu-latest
+        # Map a step output to a job output
+        outputs:
+            gocd: ${{ steps.changes.outputs.gocd }}
+        steps:
+          - uses: actions/checkout@v3
```

Comment:
Ideally we should use shas rather than versions. Also, in this case v4 is newer and should be used instead of v3.

Reply:
Just copied from your PR 😏